src/server.h

```diff
@@ -119,6 +119,16 @@ struct ServerPlayingSound
 
 	SoundSpec spec;
 
+	// server processing values
+
+	// relative server time when sound has been added
+	// used for update spec.start_time when sending sound to new client
+	float start_time;
+	// hold time limit for sound sending to new clients
+	float keep_time;
```

Comment:
I am slightly worried about this being a `float` due to potential precision issues (quick back of the napkin math suggests after `2^24 / 60 / 60 / 24 / 30`, so approximately seven months, you will lose a second of precision). I'd bump this (and `m_playing_sound_time` ) to a `double`, or perhaps even a [`std::chrono::duration`](https://en.cppreference.com/w/cpp/chrono/duration) just to be sure. I'm also not sure how accurate `dtime` is, so I would consider using the system clock via `std::chrono` to get more accurate timestamps. Thoughts?

(Side note: This is tangentially related to the more general question of how we want to synchronize client and server.)

Reply:
In this case, I would suggest the use of `m_game_time` and `m_game_time_fraction_counter`  from the `ServerEnvironment` class.

Does it look fine for you?